plugins/module_utils/utils.py

```diff
@@ -36,12 +37,26 @@ def do_differ(current, desired):
         return True
 
     for key, value in desired.items():
-        if value != current.get(key):
+        current_value = current.get(key)
+        # list elements are of the form {"type": "a", "name": "b"}
+        if (isinstance(value, list) and
+                all([isinstance(i, dict) and "name" in i and "type" in i for i in value])):
+            if type_name_dicts_differ(value, current_value):
+                return True
+        elif value != current_value:
```

Comment:
I think that we should leave keep the default comparator as simple as possible. Instead of extending this comparator to handle all use cases, we can create specialized ones and then pass them to the `sync` function as an additional argument (that we need to add, of course ;).

Sync method declaration would look something like this:

    def sync(state, client, path, payload, check_mode, comparator=do_differ):

We can then hide all role-binding-specific stuff next to the main function or in a separate role utils module. This last one might be a better approach since cluster role bindings might need similar functionality.

Does this sound sane?

Reply:
Separate role utils module might come in handy for `role` and `cluster_role` special comparators too.

Reply:
I agree, thanks!